x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeListener.java

```diff
@@ -188,6 +190,45 @@ ActionListener<Void> acquireAvoid() {
         });
     }
 
+    /**
+     * Should we ignore a failure from the remote cluster due to skip_unavailable=true setting?
+     */
+    boolean shouldIgnoreRemoteErrors(@Nullable String computeClusterAlias) {
+        return computeClusterAlias != null
+            && esqlExecutionInfo.isCrossClusterSearch()
+            && computeClusterAlias.equals(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY) == false
+            && runningOnRemoteCluster() == false
+            && esqlExecutionInfo.isSkipUnavailable(computeClusterAlias);
+    }
+
+    /**
+     * Marks the cluster as PARTIAL and adds the exception to the cluster's failures record.
+     * Currently, additional failures are not recorded, TODO: check if this should be the case.
+     */
+    void markAsPartial(String computeClusterAlias, Exception e) {
```

Comment:
Should we add this to the EsqlCCSUtils or EsqlComputeCCSUtils class that I recommended in the previous comment? That would be in keeping with Costin's request to mix as little detailed CCS "housekeeping" methods into the core ESQL flows as possible.

But I also see in the code in ComputeService that that requires passing EsqlExecutionInfo further down the call stack, so it's a tradeoff.

Reply:
This one needs ComputeListener because it holds the execution info. I could move the logic to another class but there's really not that much of the logic here...